simdjson/errors.h

```diff
@@ -1,4 +1,43 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include "simdjson.h"
+
 #ifndef _PY_SIMDJSON_ERRORS
 #define _PY_SIMDJSON_ERRORS
     void simdjson_error_handler();
+    template<typename T>
+    void * flatten_array(simdjson::dom::array src, size_t *size);
+
+    template<typename T>
+    void _flatten_array(T ** buffer, simdjson::dom::array src) {
+        for (simdjson::dom::element field : src) {
+            if (field.type() == simdjson::dom::element_type::ARRAY) {
+                _flatten_array<T>(buffer, field);
+            } else {
+                **buffer = (T)field;
+                (*buffer)++;
+            }
+        }
+    }
+
+    template<typename T>
+    void * flatten_array(simdjson::dom::array src, size_t *size) {
```

Comment:
@lemire is there any way to access the tape in the current DOM api? The tape is already flat, being able to iterate over the raw tape would make a non-recursive flatten trivial. Seems like the tape is always marked private now.